### PR TITLE
Docs: Consistent terms for client/server statistics; other improvements

### PR DESCRIPTION
**Pull request by andy-maier**
See commit message.

Resolved comments.

### CODE REVIEW

File: docs/pywbemcli/cmdshelp.rst [hunk 1]
```diff
@@ -2255,12 +2253,12 @@ Help text for ``pywbemcli statistics`` (see :ref:`statistics command group`):
...
-      server-off   Disable statistics on current server.
-      server-show  Display statistics gathered by server.
-      show         Display statistics managed by pywbemcli.
-      status       Show statistics enabled status for client and server.
+      reset        Reset client maintained statistics.
+      server-on    Enable server maintained statistics.
+      server-off   Disable server maintained statistics.
+      server-show  Display server maintained statistics.
+      show         Display client maintained statistics.
+      status       Show enablement status of client and server maintained statistics.
```
Comment (KSchopmeyer) [inline docs/pywbemcli/cmdshelp.rst:2261]: Can we change the word enablement to "enabled".

File: docs/pywbemcli/cmdshelp.rst [hunk 2]
```diff
@@ -2302,17 +2301,18 @@ Help text for ``pywbemcli statistics server-off`` (see :ref:`statistics server-o
...
-      This command turns off the gathering of statistics in the WBEM server and the return of server response times for
-      inclusion in pywbemcli maintained statistics. The gathering of server statistics and the returning of server
-      response times may not be implemented by all WBEM servers.
+      This command deactivates the gathering of statistics in the current WBEM server and also deactivates the returning
+      of server response times in the CIM-XML response for inclusion in the client maintained statistics, by setting the
+      'GatherStatisticalData' property to False in the 'CIM_ObjectManager' instance for the WBEM server.
 
-      This may fail if the server does not manage statistics or does not allow a client to modify the state of statistics
-      gathering.
+      This command will fail if the server does not support gathering statistics or does not allow a client to modify the
```
Comment (KSchopmeyer) [inline docs/pywbemcli/cmdshelp.rst:2310]: Can we change "command will fail' to 'command fails'

File: docs/pywbemcli/cmdshelp.rst [hunk 3]
```diff
@@ -2332,17 +2332,18 @@ Help text for ``pywbemcli statistics server-on`` (see :ref:`statistics server-on
...
-      This command  activates the gathering of statistics in the WBEM server and the return of server response times for
-      inclusion in pywbemcli maintained statistics. The gathering of server statistics and the returning of server
-      response times may not be implemented by all WBEM servers.
+      This command activates the gathering of statistics in the current WBEM server and also activates the returning of
+      server response times in the CIM-XML response for inclusion in the client maintained statistics, by setting the
+      'GatherStatisticalData' property to True in the 'CIM_ObjectManager' instance for the WBEM server.
 
-      This may fail if the server does not manage statistics or does not allow a client to modify the state of statistics
-      gathering.
+      This command will fail if the server does not support gathering statistics or does not allow a client to modify the
```
Comment (KSchopmeyer) [inline docs/pywbemcli/cmdshelp.rst:2341]: will fail to fails

File: docs/pywbemcli/cmdshelp.rst [hunk 4]
```diff
@@ -2362,17 +2363,15 @@ Help text for ``pywbemcli statistics server-show`` (see :ref:`statistics server-
...
-      ```statistics server-on```) for the current WBEM server.
+      Display server maintained statistics.
 
-      These statistics are independent of the statistics gathered by the pywbemcli client and displayed for example with
-      the command ``statistics show``.
+      Retrieve and display the statistics gathered by the current WBEM server. This requires statistics gathering to be
+      enabled on the server (see 'statistics server-on' command).
 
-      Presents a table of the gathered statistical information.
+      This command will fail if the server does not support gathering statistics.
```
Comment (KSchopmeyer) [inline docs/pywbemcli/cmdshelp.rst:2371]: will fail to fails

File: docs/pywbemcli/commands.rst [hunk 5]
```diff
@@ -1575,18 +1575,19 @@ environment).
...
+separately by WBEM servers that support this. There are multiple components to
 statistics gathering and reporting in pywbemcli:
 
-1.  Pywbemcli gathers and maintains statistics on CIM-XML operations it executes
-    against a WBEM server (or mock environment). These client statistics can be
-    displayed either after each pywbemcli command if the ``--timestats`` / ``-T``
-    general option is used, or in interactive mode with the ``statistics show``
-    command.
+1.  Pywbemcli gathers and maintains statistics on WBEM operations it executes
+    against a WBEM server (or mock environment). These client maintained
```
Comment (KSchopmeyer) [inline docs/pywbemcli/commands.rst:1583]: how about changing these to the "The client maintained"

File: pywbemtools/pywbemcli/_cmd_statistics.py [hunk 6]
```diff
@@ -104,20 +102,21 @@ def statistics_reset(context):
...
-    Enable statistics on current server.
+    Enable server maintained statistics.
 
-    This command  activates the gathering of statistics in the WBEM
-    server and the return of server response times for inclusion in
-    pywbemcli maintained statistics. The gathering of server statistics and the
-    returning of server response times may not be implemented by all WBEM
-    servers.
+    This command activates the gathering of statistics in the current
+    WBEM server and also activates the returning of server response times in
```
Comment (KSchopmeyer) [inline pywbemtools/pywbemcli/_cmd_statistics.py:108]: Probably don't need the work "also"

File: pywbemtools/pywbemcli/_cmd_statistics.py [hunk 7]
```diff
@@ -104,20 +102,21 @@ def statistics_reset(context):
...
+    the CIM-XML response for inclusion in the client maintained statistics,
+    by setting the 'GatherStatisticalData' property to True in the
+    'CIM_ObjectManager' instance for the WBEM server.
 
-    This may fail if the server does not manage statistics or does not allow
-    a client to modify the state of statistics gathering.
+    This command will fail if the server does not support gathering statistics
```
Comment (KSchopmeyer) [inline pywbemtools/pywbemcli/_cmd_statistics.py:113]: will fail to fails.

File: pywbemtools/pywbemcli/_cmd_statistics.py [hunk 8]
```diff
@@ -129,20 +128,21 @@ def statistics_server_on(context):
...
-    Disable statistics on current server.
+    Disable server maintained statistics.
 
-    This command turns off the gathering of statistics in the WBEM
-    server and the return of server response times for inclusion in
-    pywbemcli maintained statistics. The gathering of server statistics and the
-    returning of server response times may not be implemented by all WBEM
-    servers.
+    This command deactivates the gathering of statistics in the current
+    WBEM server and also deactivates the returning of server response times in
```
Comment (KSchopmeyer) [inline pywbemtools/pywbemcli/_cmd_statistics.py:134]: probably the also not needed.  the and covers it

File: pywbemtools/pywbemcli/_cmd_statistics.py [hunk 9]
```diff
@@ -155,18 +155,16 @@ def statistics_server_off(context):
...
+    Display server maintained statistics.
 
-    These statistics are independent of the statistics gathered by the pywbemcli
-    client and displayed for example with the command ``statistics show``.
+    Retrieve and display the statistics gathered by the current WBEM server.
+    This requires statistics gathering to be enabled on the server (see
+    'statistics server-on' command).
 
-    Presents a table of the gathered statistical information.
+    This command will fail if the server does not support gathering statistics.
```
Comment (KSchopmeyer) [inline pywbemtools/pywbemcli/_cmd_statistics.py:164]: will fail to fails.

**Review by KSchopmeyer — COMMENTED** [review]
Some documentation editorial comments and Approve

File: docs/pywbemcli/cmdshelp.rst (hunk 1 above)
Comment (andy-maier) [reply docs/pywbemcli/cmdshelp.rst:2261]: DONE

File: docs/pywbemcli/cmdshelp.rst (hunk 2 above)
Comment (andy-maier) [reply docs/pywbemcli/cmdshelp.rst:2310]: DONE

File: docs/pywbemcli/cmdshelp.rst (hunk 3 above)
Comment (andy-maier) [reply docs/pywbemcli/cmdshelp.rst:2341]: DONE

File: docs/pywbemcli/cmdshelp.rst (hunk 4 above)
Comment (andy-maier) [reply docs/pywbemcli/cmdshelp.rst:2371]: DONE

File: docs/pywbemcli/commands.rst (hunk 5 above)
Comment (andy-maier) [reply docs/pywbemcli/commands.rst:1583]: DONE

File: pywbemtools/pywbemcli/_cmd_statistics.py (hunk 7 above)
Comment (andy-maier) [reply pywbemtools/pywbemcli/_cmd_statistics.py:113]: DONE

File: pywbemtools/pywbemcli/_cmd_statistics.py (hunk 9 above)
Comment (andy-maier) [reply pywbemtools/pywbemcli/_cmd_statistics.py:164]: DONE

File: pywbemtools/pywbemcli/_cmd_statistics.py (hunk 8 above)
Comment (andy-maier) [reply pywbemtools/pywbemcli/_cmd_statistics.py:134]: We describe the command to activate and deactivate the server statistics. From that perspective, the enabling of the returning of server response time (which becomes part of the client statistics) is not what one would expect and is more like a side effect. That's why I used 'also'. NOT CHANGED.

File: pywbemtools/pywbemcli/_cmd_statistics.py (hunk 6 above)
Comment (andy-maier) [reply pywbemtools/pywbemcli/_cmd_statistics.py:108]: NOT CHANGED. See other occurrence of this.

Comment (andy-maier) [reply pywbemtools/pywbemcli/_cmd_statistics.py:108]: We agreed in the call today this is ok to go.

File: pywbemtools/pywbemcli/_cmd_statistics.py (hunk 8 above)
Comment (andy-maier) [reply pywbemtools/pywbemcli/_cmd_statistics.py:134]: We agreed in the call today this is ok to go.

**Pull request merged**